packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts

```diff
@@ -0,0 +1,182 @@
+ * @param env - server environment
+ * @returns Access Token from Authorization server
+ */
+export async function authorizeOIDC(
+  jwtToken: string,
+  env: Env,
+): Promise<AccessToken> {
+  const grantType = 'urn:ietf:params:oauth:grant-type:jwt-bearer';
+  const headers = new Headers({
+    'Content-Type': 'application/x-www-form-urlencoded',
+  });
+
+  const urlEncodedBody = new URLSearchParams();
+  urlEncodedBody.append('grant_type', grantType);
+  urlEncodedBody.append('client_id', getOidcClientId(env));
+  urlEncodedBody.append('assertion', jwtToken);
+
+  try {
+    const response = await fetch(OIDC_TOKEN_URL(env), {
+      method: 'POST',
+      headers,
+      body: urlEncodedBody.toString(),
+    });
+
+    if (!response.ok) {
+      const responseBody = (await response.json()) as {
+        error_description: string;
+        error: string;
+      };
+      throw new Error(
+        `HTTP error: ${responseBody.error_description}, error code: ${responseBody.error}`,
+      );
+    }
+
+    const accessTokenResponse = await response.json();
+    return {
+      accessToken: accessTokenResponse.access_token,
+      expiresIn: accessTokenResponse.expires_in,
+      obtainedAt: Date.now(),
+    };
+  } catch (e) {
+    /* istanbul ignore next */
+    const errorMessage =
+      e instanceof Error ? e.message : JSON.stringify(e ?? '');
+    throw new SignInError(`unable to get access token: ${errorMessage}`);
+  }
+}
+
+type Authentication = {
+  token: string;
+  expiresIn: number;
+  profile: UserProfile;
+};
+/**
+ * Service to Authenticate/Login a user via SIWE or SRP derived key.
+ *
+ * @param rawMessage - raw message for validation when authenticating
+ * @param signature - signed raw message
+ * @param authType - authentication type/flow used
+ * @param env - server environment
+ * @returns Authentication Token
+ */
+export async function authenticate(
+  rawMessage: string,
+  signature: string,
+  authType: AuthType,
+  env: Env,
+): Promise<Authentication> {
+  const authenticationUrl = getAuthenticationUrl(authType, env);
+
+  try {
+    const response = await fetch(authenticationUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        signature,
+        raw_message: rawMessage,
+      }),
+    });
+
+    if (!response.ok) {
+      const responseBody = (await response.json()) as ErrorMessage;
+      throw new Error(
+        `${authType} login HTTP error: ${responseBody.message}, error code: ${responseBody.error}`,
+      );
+    }
+
+    const loginResponse = await response.json();
+    return {
+      token: loginResponse.token,
+      expiresIn: loginResponse.expires_in,
+      profile: {
+        identifierId: loginResponse.profile.identifier_id,
+        metaMetricsId: loginResponse.profile.metametrics_id,
+        profileId: loginResponse.profile.profile_id,
+      },
+    };
+  } catch (e) {
+    /* istanbul ignore next */
```

Comment:
These are for test coverage. It is used in other packages.

This is valid since most cases the error receive with be an `Error` instance.